ci-tests/riscof/spike/riscof_spike.py

```diff
@@ -0,0 +1,179 @@
+import os
```

Comment:
My point is, the files in this directory, especially this one, have absolutely nothing to do with the Sail model. They should not be copied to every project/repository/whatever that needs to use Spike, because that quickly becomes an unmaintainable mess where everybody has a differently-outdated version, and you end up with O(n^2) of these files rather than O(n) due to the cross-product between reference and DUT. There should be some way to only have one copy of "how do I build and run tests for platform X", and the most obvious (to me) way to achieve that is to put that code in platform X's repository at a stable published location.

Reply:
I agree, but my understanding is the current RISCOF setup essentially expects the user to bring along a certain amount of boilerplate for both the thing you are testing and the thing you are testing against. I think we need to be careful that we review this PR as an attempt to improve our lack of testing by integrating RISCOF rather than look at larger issues in the RISC-V ecosystem that may not be easily actionable by the PR author.